### ISSUE TITLE
[New rule] Avoid using var

### ISSUE THREAD

**Issue by mdeandrea-mrmilu** [dart-code-checker/dart-code-metrics#1203]
### Rule details

This rule should show warning when use `var` on code.
The style guide for Flutter indicate to [avoid using var](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic) on code, so that will be a nice rule to have

### What type of rule is this?

Warns about a potential problem

### Example code

```dart
BAD:

var x = 10; // LINT

String concat(int a, int b) {
  var c = 3; // LINT
  return a + b + c;
}


GOOD:

int x = 10;

final x = 10;

String concat(int a, int b) {
  int c = 3; // or final c = 3;
  return a + b + c;
}
```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.
